### PR TITLE
Add phase2 function to MSD

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

Combines steps 3 through 5 into `phase2()`.

A subsequent PR will also include `updateDynamicConfig()` in `phase2`.

Fixes CLI-3790

### CODE REVIEW

**Comment by maurelian** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5359** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5359" target="_blank">Graphite</a> 
    * **PR #5377** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5377" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5377?utm_source=stack-comment).

File: packages/contracts-bedrock/contracts/deployment/SystemDictator.sol [hunk 1]
```diff
@@ -418,6 +418,15 @@ contract SystemDictator is OwnableUpgradeable {
         step2();
     }
 
+    /**
+     * @notice Calls the next remaings steps of the migration process.
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/deployment/SystemDictator.sol:422]:
```suggestion
     * @notice Calls the remaining steps of the migration process.
```

**Review by refcell — CHANGES_REQUESTED** [review]
Small changes otherwise lgtm pending ci fixes

File: packages/contracts-bedrock/deploy/021-SystemDictatorSteps-2.ts [hunk 2]
```diff
@@ -225,17 +170,59 @@ const deployFn: DeployFunction = async (hre) => {
...
+
+      // Step 4 checks
+      await assertContractVariable(AddressManager, 'owner', ProxyAdmin.address)
+
+      assert(
+        (await L1StandardBridgeProxy.callStatic.getOwner({
+          from: ethers.constants.AddressZero,
+        })) === ProxyAdmin.address
+      )
+
```
Comment (clabby) [inline packages/contracts-bedrock/deploy/021-SystemDictatorSteps-2.ts:224]: Should probably add a check for the ownership of the `L1ERC721Bridge` here as well.

**Review by clabby — COMMENTED** [review]
LGTM, will approve once linting is fixed and the `L1ERC721Bridge` check is added.

**Review by clabby — APPROVED** [review]
lgtm

**Review by refcell — APPROVED** [review]

**Pull request merged**